### DISCUSSION TITLE
Option to exclude commits from a certain user from status checks

### DISCUSSION THREAD

**Discussion by gertjanschouten** [community/community#46610] [Actions]
### Select Topic Area

Product Feedback

### Body

Hello!

I think it's a good idea to add an option to the Repository Settings that enables you to exclude commits done by a certain user from being taken into account when doing the status checks. The reason for this:

We have a GitHub Actions workflow that gets triggered on push.

We also have branch protection rules that wait for check results, which are just various phases of the workflow, like the build and the deploy to the test environment.

As part of our GitHub Actions workflow, certain files get generated and committed into the repository. This does not trigger the same workflow again, because the commit is done with the GITHUB_TOKEN. This is _desired_ and _intentional_. We don't want another workflow triggered.

However... The branch protection rules now aren't satisfied with the check results anymore, because the check results were from a workflow that did not include the commit from the workflow itself. The branch protection rules are only satisfied with check results that were reported from a workflow that includes all commits, including the commit from the workflow itself.

This is a problem. Whenever the workflow generates and commits files, the branch protection rules keep waiting indefinitely for check results to be reported.

The issue is described in more detail at https://support.github.com/ticket/personal/0/1975812 and is simulated here: https://github.com/gertjanschouten/GA1975812_demonstration/pull/1

This PR keeps waiting on the "build" status indefinitely. Ignoring commits from the GitHub Actions user from being taken into account when doing the status checks would resolve this.

**Comment by josafatburmeister** [comment]
Are there any ideas on how to work around this issue?
